packages/salesforcedx-vscode-core/src/metadataSupport/metadataHoverProvider.ts

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'node:path';
+import * as vscode from 'vscode';
+import { MetadataDocumentationService } from './metadataDocumentationService';
+
+/**
+ * Provides hover documentation for Salesforce metadata types in XML files
+ */
+export class MetadataHoverProvider implements vscode.HoverProvider {
+  private documentationService: MetadataDocumentationService;
+
+  constructor() {
+    this.documentationService = new MetadataDocumentationService();
+  }
+
+  /**
+   * Initialize the hover provider by loading metadata documentation
+   */
+  public async initialize(): Promise<void> {
+    await this.documentationService.initialize();
+  }
+
+  /**
+   * Provide hover information for metadata types and their internal fields
+   */
+  public async provideHover(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+    _token: vscode.CancellationToken
+  ): Promise<vscode.Hover | null> {
+    // Only provide hover for XML files that are likely metadata files
+    if (!this.isMetadataFile(document)) {
+      return null;
+    }
+
+    const wordRange = document.getWordRangeAtPosition(position);
+    if (!wordRange) {
+      return null;
+    }
+
+    const word = document.getText(wordRange);
+    const line = document.lineAt(position.line);
+    const lineText = line.text;
+
+    // First, try to get metadata type documentation
+    const metadataType = this.extractMetadataType(lineText, word, position.character);
+    if (metadataType) {
+      const documentation = this.documentationService.getDocumentation(metadataType);
+      if (documentation) {
+        const markdownContent = new vscode.MarkdownString();
+        markdownContent.appendCodeblock(metadataType, 'xml');
+        markdownContent.appendMarkdown(documentation.description);
+
+        if (documentation.fields && documentation.fields.length > 0) {
+          markdownContent.appendMarkdown('\n\n**Key Fields:**\n');
```

Comment:
oh, maybe this is the reason for needing a custom hover?  because just the `documentation` is too big/unusable?

My preference is that the PR body explains some of the non-obvious "why"